### DISCUSSION TITLE
Can a physics interaction be replayed backward?

### DISCUSSION THREAD

**Discussion by quacrobat** [pmndrs/react-three-rapier#186] [Q&A] [answered]
Hi, considering Rapier is deterministic (if this is relevant), is it possible to save the trajectories of different scene objects and then invert the animation? For example a ball bounces down then comes to a stop, in inverted time, it starts from a resting position on the floor then starts bouncing higher and higher until it goes back to initial position.
Thanks

**Comment by wiledal** [comment, marked as answer]
Hey,

There's no way to play the simulation in reverse, but you could always use the API to record the RigidBody's transform and velocity, to more or less return it to a previous state.

For instance:

```ts
const timeline = []
const recording = true

const Object = () => {
  const ref = useRef<RigidBodyApi>(null)
  
  useFrame(() => {
    if (recording && ref.current) {
      const translation = ref.current.translation()
      const rotation = ref.current.rotation()
      const linvel = ref.current.linvel()
      
      const frame = { translation, rotation, linvel }
      timeline.push(frame) 
    }
  })
  
  return <RigidBody ref={ref} ... />
}
```

You can use those values to "play" the timeline backwards.
You can also pause the simulation by setting `<Physics paused />`.

**Reply by quacrobat** [reply]
super, thank you!
